### PR TITLE
Remove Deprecations

### PR DESCRIPTION
**Pull request by dvesh3**
## Changes in this pull request  
Resolves #

### <samp>🤖 Generated by Copilot at 83c7fa5</samp>

This pull request removes some deprecated code and unused files from the AdminBundle and the Asset model, and simplifies some internal methods in the Navigation and Page classes. It also updates the documentation with some upgrade notes for Pimcore 11.

### <samp>🤖 Generated by Copilot at 83c7fa5</samp>

> _`Pimcore` evolves_
> _Removing old code and docs_
> _Winter of cleanup_

### <samp>🤖 Generated by Copilot at 83c7fa5</samp>

*  Remove deprecated and unused code related to custom layouts, username, and asset types ([link](https://github.com/pimcore/pimcore/pull/14897/files?diff=unified&w=0#diff-223386524f4ff18dc1e37479f7e783a5e463cce8417be7519db1a05185549f1aL598-L612), [link](https://github.com/pimcore/pimcore/pull/14897/files?diff=unified&w=0#diff-3d05f5a063118bc4fe99a473a641413060d30af086a754108ba3e225aaa149c6L44-L51), [link](https://github.com/pimcore/pimcore/pull/14897/files?diff=unified&w=0#diff-867a334e4cc9a097b215afcb2b1f4949910cb2f2103a2cd997d8699d4d414fc9L70-R71))
*  Delete files for the obsolete AdminSessionHandler and AdminSessionListener classes ([link](https://github.com/pimcore/pimcore/pull/14897/files?diff=unified&w=0#diff-f41984f15a4bc095affc33fc15b0bb41482fea9ab1e10f5a4a514b46eaca84b1), [link](https://github.com/pimcore/pimcore/pull/14897/files?diff=unified&w=0#diff-6af7519629d1db7b5f0af1b0ec0c542dcdee01eb48ed13c73df7f14c4835d45c), [link](https://github.com/pimcore/pimcore/pull/14897/files?diff=unified&w=0#diff-849b16107e73d7978f2585efdf29f20cc218e5dcee6209adeb3682b0cff1170a))
*  Add upgrade notes for Pimcore 11 to the documentation (`README.md`) ([link](https://github.com/pimcore/pimcore/pull/14897/files?diff=unified&w=0#diff-375aded12dfb868b84d5b8c27d1aed2358680a27b683e8a985f1b85193dd79aaR316-R318))
*  Simplify and mark as internal the methods for setting and getting the default page type in the `Page` class (`Page.php`) ([link](https://github.com/pimcore/pimcore/pull/14897/files?diff=unified&w=0#diff-55e361c39be91eefa9510517e5ed25379f74564897599ac16d47de2011d50bbfL1116-R1127))

### CODE REVIEW

File: lib/Navigation/Page.php [hunk 1]
```diff
@@ -1113,24 +1113,18 @@ protected static function _normalizePropertyName(string $property): string
     }
```
Comment (kingjia90) [inline lib/Navigation/Page.php:1109]: by looking at
https://github.com/pimcore/pimcore/blob/83c7fa51381e4bf72b52c70eb91db1cae0cb0ba0/lib/Navigation/Page.php#L1101-L1110
it seems that `_normalizePropertyName` should be also `@internal`

the phpdoc block can be slimmed, not quite useful

**Review by kingjia90 — CHANGES_REQUESTED** [review]

File: lib/Navigation/Page.php (hunk 1 above)
Comment (martineiber) [reply lib/Navigation/Page.php:1109]: Than we should also mark the _sort() method also as internal?

https://github.com/pimcore/pimcore/blob/230610923ee36ab370261fd13a84619be0162ad8/lib/Navigation/Container.php#L65-L75

File: doc/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md [hunk 2]
```diff
@@ -313,6 +313,9 @@ pimcore:
 }
 ```
 - Removed the following methods from `Pimcore\File`: `mkdir`, `put`, `getFileExtension`, `setDefaultMode`, `getDefaultMode`, `setDefaultFlags` and `rename`. 
+- Removed deprecated method `Pimcore\Bundle\AdminBundle\Security\User::getUsername()`, use `getIdentifier()` instead.
```
Comment (martineiber) [inline doc/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md:316]:
```suggestion
- Removed deprecated method `Pimcore\Bundle\AdminBundle\Security\User::getUsername()`, use `getIdentifier()` instead.
- Removed deprecated property `Pimcore\Model\Asset::$types`, use `getTypes()` instead
```

**Review by martineiber — CHANGES_REQUESTED** [review]

File: lib/Navigation/Page.php (hunk 1 above)
Comment (dvesh3) [reply lib/Navigation/Page.php:1109]: yes, we should mark all methods internal falling under this comment https://github.com/pimcore/pimcore/blob/53449686033cdb9dd10f5c64c9636b8c725e2a35/lib/Navigation/Page.php#L1101

File: doc/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md [hunk 3]
```diff
@@ -313,6 +313,10 @@ pimcore:
...
+- Removed deprecated property `Pimcore\Model\Asset::$types`, use `getTypes()` instead
+- Removed AdminSessionHandler and AdminSessionListener. The session is now handled by Symfony.
+- Methods `Pimcore\Navigation::setDefaultPageType` and `Pimcore\Navigation::getDefaultPageType` have been marked as internal.
```
Comment (martineiber) [inline doc/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md]:
```suggestion
- Methods `Pimcore\Navigation::setDefaultPageType`, `Pimcore\Navigation::getDefaultPageType`, `Pimcore\Navigation\Container::_sort() and `Pimcore\Navigation\Page::_normalizePropertyName()` have been marked as internal.
```

**Review by martineiber — CHANGES_REQUESTED** [review]

**Review by martineiber — APPROVED** [review]

**Pull request merged**